openedx/core/djangoapps/content/search/api.py

```diff
@@ -0,0 +1,486 @@
+"""
+Content index and search API using Meilisearch
+"""
+from __future__ import annotations
+
+import logging
+import time
+from contextlib import contextmanager
+from datetime import datetime, timedelta, timezone
+from functools import wraps
+from typing import Callable, Generator
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.cache import cache
+from meilisearch import Client as MeilisearchClient
+from meilisearch.errors import MeilisearchError
+from meilisearch.models.task import TaskInfo
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import LibraryLocatorV2
+
+from openedx.core.djangoapps.content.search.documents import (
+    STUDIO_INDEX_NAME,
+    Fields,
+    meili_id_from_opaque_key,
+    searchable_doc_for_course_block,
+    searchable_doc_for_library_block
+)
+from openedx.core.djangoapps.content_libraries import api as lib_api
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+from .documents import Fields, searchable_doc_for_course_block, searchable_doc_for_library_block
+
+log = logging.getLogger(__name__)
+
+User = get_user_model()
+
+STUDIO_INDEX_NAME = "studio_content"
+
+if hasattr(settings, "MEILISEARCH_INDEX_PREFIX"):
+    INDEX_NAME = settings.MEILISEARCH_INDEX_PREFIX + STUDIO_INDEX_NAME
+else:
+    INDEX_NAME = STUDIO_INDEX_NAME
```

Comment:
I find `STUDIO_INDEX_NAME` and `INDEX_NAME` a bit confusing here.

Perhaps rename `STUDIO_INDEX_NAME` to `STUDIO_INDEX_SUFFIX` and rename `INDEX_NAME` to `STUDIO_INDEX_NAME` because what you've called `INDEX_NAME` _is_ the final name of the studio index.

Reply:
Done: 1362a4633d68f18a30d7f347b9c568c61482bf1c

For now, I'm using the constants inside the functions, but we can easily use the index name as a parameter in the future.